### PR TITLE
Introduction to python docs converted to Sphinx

### PR DESCRIPTION
**Pull request by keeeto**
**Description of work.**
This PR continues the conversion of docs from media wiki to Sphinx. In this instance the `Introduction to Python` has been copied over.  

**To test:**

* Build the `docs-html` target.
* Open `html/index.html`
* Navigate to `Training` then `Introduction to Python`
* Check that the pages have built and links work
* Test some of the examples to see if they work
 
*There is no associated issue.*

*This does not require release notes* because it is not a change to the code.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/tutorials/introduction_to_python/working_with_functions.rst [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+   xsq, ysq = square(2,3)
+   print(xsq)  # Prints 4
+   print(ysq)  # Prints 9  
+   # Tuple has vanished!
+
+.. raw:: mediawiki
+
+   {{SlideNavigationLinks|Working_With_Functions|Introduction_To_Python|Basic_Python_Exercises_3-4}}
+
+`Category:Tested Examples <Category:Tested_Examples>`__
```
Comment (peterfpeterson) [inline docs/source/tutorials/introduction_to_python/working_with_functions.rst]: Generally URL links aren't used in sphinx. Use a `:ref:` to the right page.

**Comment by peterfpeterson** [comment]
Two big questions
1. `{{SlideNavigationLinks|Getting_to_know_Python|Introduction_To_Python|Using_Modules}}` doesn't appear to do anything, but it is in every page (random sampling). What is it supposed to do?
2. If these are "tested examples" then shouldn't be actually tested automatically rather than code-blocks?

**Comment by keeeto** [comment]
I am marking this In Progress while I make the code testable - a good point from @peterfpeterson .

**Comment by keeeto** [comment]
This went stale so I have restarted again on a PR #24380.

**Pull request closed without merging**